lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,551 @@
+            name=dict(type='str', required=True),
+            resource_group=dict(type='str', required=True),
+            properties=dict(type='dict'),
+            location=dict(type='str'),
+            status=dict(type='str', default='Enabled', choices=['Enabled', 'Disabled']),
+            state=dict(type='str', default='present', choices=['present', 'absent'])
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.properties = None
+        self.location = None
+        self.status = None
+        self.state = None
+
+        self.results = dict(
+            changed=False,
+            contains_endpoints=False,
+            state=dict())
+
+        super(AzureRMTrafficManager, self).__init__(self.module_arg_spec,
+                                                    supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        # Collect all the tags and add them to the attributes
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        # Before we proceed, verify that the resource group exists.
+        try:
+            self.get_resource_group(self.resource_group)
+
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
+
+        # Initialize the Ansible return
+        results = dict()
+        changed = False
+        current_traffic_manager = self.get_traffic_manager_profile(
+            self.resource_group,
+            self.name)
+
+        results = self.traffic_manager_to_dict(
+            current_traffic_manager,
+            self.resource_group,
+            self.name, self.location,
+            self.properties)
+
+        try:
+            if self.state == 'present':
+                self.log('Fetching traffic manager {0}'.format(self.name))
+                # dry-run.
+                if self.check_mode:
+                    # if the current_traffic_manager is empty it means we will
+                    # create a new profile. Return a profile object to the user
+                    if not bool(current_traffic_manager):
+                        self.results['state'] = results
+                    # if the profile exists, compare and report back the differences
+                    else:
+                        self.results['state'] = self.compare_profiles(
+                            current_traffic_manager,
+                            self.properties)
+                    return self.results
+
+                # Not in check_mode. This will create the profile.
+                else:
+                    if not bool(current_traffic_manager):
+                        # profile doesn't exist. We will created it.
+                        # Traffic Manager uses DNS, we need to check if the
+                        # name used in the DNS prefix is available
+                        try:
```

Comment:
Is 'the name used in the DNS prefix' check required to create the traffic manager? Or the error is already handled by the create API? If so, I don't this this logic is needed here. If it is required, it should be include in check_mode